### ISSUE TITLE
Detection of free memory in refine script inaccurate on Mac OS

### ISSUE THREAD

**Issue by wetneb** [OpenRefine/OpenRefine#5455]
The detection of free memory in the refine script is inconsistent on Mac OS and Linux.

On Linux, this line normally prints the total amount of RAM available (although it claims to display the amount of "free memory"). On MacOS it currently relies on the following Bash command:

```
top -l 1 |grep PhysMem| awk '{print $6}'
```
That seems to output something really small (some amount of physical RAM not allocated by the OS? in any case not the amount of memory available to applications).

To be consistent with the Linux version (showing the total memory), it should instead be:
```
top -l 1 |grep PhysMem| awk '{print $2}'
```

The message should be adapted accordingly, of course.

### To Reproduce
1. run `./refine`

### Current Results

```
-------------------------------------------------------------------------------------------------
You have 29M of free memory.
Your current configuration is set to use 1024M of memory.
OpenRefine can run better when given more memory. Read our FAQ on how to allocate more memory here:
https://docs.openrefine.org/manual/installing#increasing-memory-allocation
-------------------------------------------------------------------------------------------------
```

### Expected Behavior
```
-------------------------------------------------------------------------------------------------
You have 7772M of memory.
Your current configuration is set to use 1024M of memory.
OpenRefine can run better when given more memory. Read our FAQ on how to allocate more memory here:
https://docs.openrefine.org/manual/installing#increasing-memory-allocation
-------------------------------------------------------------------------------------------------
```

### Versions
 - Operating System: Mac OS X
 - OpenRefine: 3.7-SNAPSHOT

### Additional context

Encountered on a Mac OS machine given by @trnstlntk. Thank you Sandra!

**Comment by ostephens** [comment]
The command for macOS also assumes that the result will be given in Mb and removes the trailing "M" from the value found:
`freeRam=$(top -l 1 | grep PhysMem | awk '{print $6}' | tr -d M)`

The "M" is then added back when printing out the message for the user:
`echo You have "$freeRam"M of free memory`

In the case of the memory available being in some unit other than Mb, this leads to incorrectly stated memory. e.g. on my system:
`You have 15GM of free memory.`

When obviously a "GM" is not a valid measure of memory

**Comment by tfmorris** [comment]
I don't think this is a good first issue until the desire behavior is better specified. The top level question to answer is "What's the most relevant figure for the user - free memory, as currently specified, or something else like total or available memory?"

On my 16GiB Mac, the output of `top -l 1 |grep PhysMem` is

`PhysMem: 16G used (3125M wired), 198M unused.`

so the existing code is correct (although very slow) for getting the amount of free memory ie the label and the code match. 

On an 8GiB Linux machine `free -m` produces

```
              total        used        free      shared  buff/cache   available
Mem:           7857         658        4162           0        3036        6872
```

and the current code returns 7857, which is clearly wrong. 4162 is the number which matches the label, although, arguably, the amount of **available** memory, 6872, would be better (with an appropriate change in the labelling, of course). ie `free -m | grep Mem: | awk '{print $7}'`

The answers to this StackOverflow [question](https://apple.stackexchange.com/questions/4286/is-there-a-mac-os-x-terminal-version-of-the-free-command-in-linux-systems) outline some Mac equivalents. It looks like either `vm_stat` or `memory_pressure` would be better ways to get relevant information on the Mac, although, due to the vagaries of Mach, the "available" memory calculation is a little complicated. I *think* it would be `free + speculative + inactive` (plus the appropriate math to convert from 4K pages to MiB or the desired output unit.

The output from `memory_pressure` on my 16GB system is 
```
The system has 17179869184 (4194304 pages with a page size of 4096).
...
System-wide memory free percentage: 74%
```

which is 11.8GiB available and the output from `vm_stat` is
```
Mach Virtual Memory Statistics: (page size of 4096 bytes)
Pages free:                              293819.
Pages active:                           1410765.
Pages inactive:                         1358871.
Pages speculative:                        51311.
Pages throttled:                              0.
Pages wired down:                        797039.
Pages purgeable:                          16565.
...
```
which gives a total of 12.9GiB of pages not wired.

I suggest that the most interesting useful figures are the available memory, not free (or total), so both Linux & Mac implementations should change (in different ways).

**Comment by wetneb** [comment]
Absolutely! I got used to seeing the total memory on Linux so I was expecting the same on MacOS but normalizing both to the available memory also works.
